### PR TITLE
Feat/example training internlm

### PR DESCRIPTION
**Pull request by huangting4201**
## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.

### CODE REVIEW

File: internlm/train/training_internlm.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+        )
+
+    if dataset_generate_func is None:
+        train_collate_fn = partial(packed_collate_fn, packed_length=data_cfg.packed_length)
+
+    # Create the training data loader
+    train_dl = DataLoader(
+        dataset=train_ds,
+        batch_sampler=train_sampler,
+        num_workers=data_cfg.get("num_worker", 0),
```
Comment (sunpengsdu) [inline internlm/train/training_internlm.py:198]: 这里最好不要动之前那个接口，num_worker默认为0很危险

Comment (sunpengsdu) [reply internlm/train/training_internlm.py:198]: 或者这里做一个默认为4，需要在template config里补上这个参数？

Comment (huangting4201) [reply internlm/train/training_internlm.py:198]: 那这块还是保留参数传入吧

File: train.py [hunk 2]
```diff
@@ -507,8 +127,8 @@ def main(args):
     criterion = FlashGPTLMLoss(parallel_output=True, label_smoothing=label_smoothing)
 
     # initialize the train and validation data loader
-    train_dl, dataset_types = get_train_data_loader(num_worker=4)
-    val_dls = get_validation_data_loader()
+    train_dl, dataset_types = get_train_data_loader()
+    val_dls = get_validation_data_loader(logger=logger)
```
Comment (sunpengsdu) [inline train.py:131]: 为啥这个还要专门传一个logger?

**Pull request merged**